### ISSUE TITLE
Raspberry Pi 5 exec changes

### ISSUE THREAD

**Issue by glossyio** [AlexxIT/go2rtc#881]
The documentation to get the RPi 5 working with the `exec` stream needs to be updated: https://github.com/AlexxIT/go2rtc#source-exec

Currently reads
```
streams:
  picam_h264: exec:libcamera-vid -t 0 --inline -o -
  picam_mjpeg: exec:libcamera-vid -t 0 --codec mjpeg -o -
```

The RPi5 needs to have the libav codec specified, see explanation below.
```
streams:
  picam_h264: exec:rpicam-vid -t 0 --inline --libav-format h264 -o -
  picam_mjpeg: exec:rpicam-vid -t 0 --codec mjpeg --libav-format mpegts -o -
```

Explanation, per RPi engineer at https://forums.raspberrypi.com/viewtopic.php?t=360020#p2180108

> The libav codec (used on a Pi 5) tries to deduce the container/output format (different from the encoding format) based on the file name. Since you are specifying a network address as a filename, it cannot do this, so you must manually provide the output format through the --libav-format command line option.

In addition, the RPi-native `libcamera-vid` is being renamed `rpicam-vid`, per https://forums.raspberrypi.com/viewtopic.php?t=359569&sid=fee8e58134bceb78f9f5581094233a27

**Comment by AlexxIT** [comment]
Are you sure about `mjpeg` version? It's very strange to use it with `mpegts`.

**Comment by glossyio** [comment]
> Are you sure about `mjpeg` version? It's very strange to use it with `mpegts`.

Thanks for looking at this. I haven't used mjpeg only the h264, so I cannot confirm. I included it because that is what the RPi engineer said to try in their forum comment.

**Comment by AlexxIT** [comment]
I haven't seen example with mjpeg+mpegts.

**Comment by rici44** [comment]
Using go2rtc in docker compose container I am not able to use rpicam stream. 
Is [Source: Exec](https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#source-exec) usable inside docker compose container?

**Comment by glossyio** [comment]
> Using go2rtc in docker compose container I am not able to use rpicam stream. Is [Source: Exec](https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#source-exec) usable inside docker compose container?

No, `rpicam` is not available in a Docker container unless you install the library and all dependencies in the container **and then pass through the camera as a device**.  I found it easier to install/configure go2rtc on the host machine and call the rtsp stream in the container.

**Comment by rici44** [comment]
> > Using go2rtc in docker compose container I am not able to use rpicam stream. Is [Source: Exec](https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#source-exec) usable inside docker compose container?
> 
> No, `rpicam` is not available in a Docker container unless you install the library and all dependencies in the container **and then pass through the camera as a device**. I found it easier to install/configure go2rtc on the host machine and call the rtsp stream in the container.

And how do you install and configure go2rtc and if you install what container do you call rtsp stream from?

**Comment by rici44** [comment]
thx @glossyio  that works well

**Comment by jneilliii** [comment]
@glossyio wouldn't installing the codecs work on OS lite? `sudo apt install libavcodec-extra`

**Comment by glossyio** [comment]
> wouldn't installing the codecs work on OS lite? `sudo apt install libavcodec-extra`

@jneilliii - you are totally right.  Someone got that working without having to rebuild, and I guess there was no link back to this issue.  Check it out:  https://github.com/geerlingguy/3d-printing/issues/9#issuecomment-2435919505 

It wasn't just codecs install, but the whole `rpicam-apps library` .

**Comment by Tcoton** [comment]
I confirm that the command ` picam_mjpeg: exec:rpicam-vid -t 0 --codec mjpeg --libav-format mpegts -o -` does not work, it goes to a black screen.
